apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx

```diff
@@ -303,30 +320,34 @@ class ScriptEditor extends React.Component {
       dataToSave.stage_descriptions = this.state.lessonDescriptions;
     }
 
-    $.ajax({
-      url: `/s/${this.props.id}`,
-      method: 'PUT',
-      dataType: 'json',
-      contentType: 'application/json;charset=UTF-8',
-      data: JSON.stringify(dataToSave)
-    })
-      .done(data => {
-        if (shouldCloseAfterSave) {
-          navigateToHref(linkWithQueryParams(data.scriptPath));
-        } else {
-          const lessonGroups = mapLessonGroupDataForEditor(data.lesson_groups);
-
-          this.props.init(lessonGroups, this.props.levelKeyList);
-          this.setState({
-            lastSaved: Date.now(),
-            isSaving: false,
-            oldScriptText: data.lessonLevelData
-          });
-        }
+    if (this.state.isSaving) {
```

Comment:
this looks like a red flag because you are checking the state which you just set. you can't expect the call to setState to have updated yet. if you do want to wait for it to update, you can do so using the optional callback parameter to setState.